crates/bevy_ecs/src/query/iter.rs

```diff
@@ -331,17 +331,26 @@ where
         if max_size < K {
             return (0, Some(0));
         }
+        if max_size == K {
+            return (1, Some(1));
+        }
 
-        // n! / k!(n-k)! = (n*n-1*...*n-k+1) / k!
-        let max_combinations = (0..K)
-            .try_fold(1usize, |n, i| n.checked_mul(max_size - i))
-            .map(|n| {
-                let k_factorial: usize = (1..=K).product();
-                n / k_factorial
-            });
+        // binomial coefficient: (n ; k) = n! / k!(n-k)! = (n*n-1*...*n-k+1) / k!
+        // See https://en.wikipedia.org/wiki/Binomial_coefficient
+        // See https://blog.plover.com/math/choose.html for implementation
+        // It was chosen to reduce overflow potential.
```

Comment:
Note: It reduces overflow potential at least compared to the previous solution which computed `n!` and `k!(n-k)!`. This still can overflow even when the result is not an overflow. An example is provided in https://blog.plover.com/math/choose-2.html

It's also probably slower than the previous solution, because it does `k` div and `k` checked_mul, while the previous one did 1 div, `k` mul and `k` checked_mul.